overlord/devicestate/systems.go

```diff
@@ -176,3 +186,212 @@ func seededSystemFromModeenv() (*seededSystem, error) {
 	}
 	return seededSys, nil
 }
+
+func createSystemForModelFromValidatedSnaps(getInfo func(name string) (*snap.Info, bool, error), db asserts.RODatabase, label string, model *asserts.Model) (newFiles []string, dir string, err error) {
```

Comment:
nitpick: I would have the arguments in this order

model, label, db, getInfo

also I would give names to the return arguments of getInfo otherwise the bool is unclear

This merits a doc comment explaing a bit what it does concretely and the expectations on getInfo

Reply:
done